backend/controller/controller.go

```diff
@@ -1810,54 +1811,19 @@ func (s *Service) syncRoutes(ctx context.Context) (ret time.Duration, err error)
 				}
 			}
 			newRoutes[v.Module] = Route{Module: v.Module, Deployment: v.Key, Endpoint: targetEndpoint}
+			modulesByName[v.Module] = v.Schema
 		}
 	}
-	s.routes.Store(newRoutes)
-	return time.Second, nil
-}
-
-// Synchronises Service.schema from the database.
-func (s *Service) syncSchema(ctx context.Context) {
-	logger := log.FromContext(ctx)
-	modulesByName := map[string]*schema.Module{}
-	retry := backoff.Backoff{Max: time.Second * 5}
-	for {
-		err := s.watchModuleChanges(ctx, func(response *ftlv1.PullSchemaResponse) error {
-			switch response.ChangeType {
-			case ftlv1.DeploymentChangeType_DEPLOYMENT_ADDED, ftlv1.DeploymentChangeType_DEPLOYMENT_CHANGED:
-				moduleSchema, err := schema.ModuleFromProto(response.Schema)
-				if err != nil {
-					return err
-				}
-				modulesByName[moduleSchema.Name] = moduleSchema
 
-			case ftlv1.DeploymentChangeType_DEPLOYMENT_REMOVED:
-				delete(modulesByName, response.ModuleName)
-			}
+	orderedModules := maps.Values(modulesByName)
+	sort.SliceStable(orderedModules, func(i, j int) bool {
+		return orderedModules[i].Name < orderedModules[j].Name
+	})
+	combined := &schema.Schema{Modules: orderedModules}
+	s.schema.Store(ftlreflect.DeepCopy(combined))
```

Comment:
I don't think you need the DeepCopy anymore?

Reply:
good point